src/contracts/payloads/PayloadsControllerCore.sol

```diff
@@ -219,6 +215,25 @@ abstract contract PayloadsControllerCore is
 
   receive() external payable {}
 
+  /**
+   * @notice method to cancel a payload
+   * @param payloadId id of the payload that needs to be canceled
+   */
+  function _cancelPayload(uint40 payloadId) internal {
```

Comment:
why abstract the logic, if later you can call the super.cancelPayload ? its not like you are re-using the logic anywhere else apart from the child method